1.3.0 build fails w/ mariadb-devel-5.5.68

$ rpm -qa|grep -i mariadb
mariadb-libs-5.5.68-1.amzn2.0.1.aarch64
mariadb-5.5.68-1.amzn2.0.1.aarch64
mariadb-server-5.5.68-1.amzn2.0.1.aarch64
mariadb-devel-5.5.68-1.amzn2.0.1.aarch64



R version 4.3.1 (2023-06-16) -- "Beagle Scouts"
Copyright (C) 2023 The R Foundation for Statistical Computing
Platform: aarch64-unknown-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> install.packages("RMariaDB")
Installing package into '/home/xxx/R/aarch64-unknown-linux-gnu-library/4.3'
(as 'lib' is unspecified)
trying URL 'https://cloud.r-project.org/src/contrib/RMariaDB_1.3.0.tar.gz'
Content type 'application/x-gzip' length 907444 bytes (886 KB)
==================================================
downloaded 886 KB

* installing *source* package 'RMariaDB' ...
** package 'RMariaDB' successfully unpacked and MD5 sums checked
** using staged installation
Using mysql_config
Found mysql_config cflags and libs!
Using PKG_CFLAGS=-I/usr/include/mysql
Using PKG_LIBS=-L/usr/lib64/mysql -lmysqlclient -lpthread -lz -lm -ldl -lssl -lcrypto
Using PKG_PLOGR=
** libs
using C++ compiler: 'g++ (GCC) 7.3.1 20180712 (Red Hat 7.3.1-17)'
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/library/cpp11/include' -I'/usr/lib/R/library/plogr/include' -I/usr/local/include    -fPIC  -g -O2  -c DbConnection.cpp -o DbConnection.o
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/library/cpp11/include' -I'/usr/lib/R/library/plogr/include' -I/usr/local/include    -fPIC  -g -O2  -c DbResult.cpp -o DbResult.o
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/library/cpp11/include' -I'/usr/lib/R/library/plogr/include' -I/usr/local/include    -fPIC  -g -O2  -c MariaBinding.cpp -o MariaBinding.o
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/library/cpp11/include' -I'/usr/lib/R/library/plogr/include' -I/usr/local/include    -fPIC  -g -O2  -c MariaResult.cpp -o MariaResult.o
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/library/cpp11/include' -I'/usr/lib/R/library/plogr/include' -I/usr/local/include    -fPIC  -g -O2  -c MariaResultImpl.cpp -o MariaResultImpl.o
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/library/cpp11/include' -I'/usr/lib/R/library/plogr/include' -I/usr/local/include    -fPIC  -g -O2  -c MariaResultPrep.cpp -o MariaResultPrep.o
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/library/cpp11/include' -I'/usr/lib/R/library/plogr/include' -I/usr/local/include    -fPIC  -g -O2  -c MariaResultSimple.cpp -o MariaResultSimple.o
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/library/cpp11/include' -I'/usr/lib/R/library/plogr/include' -I/usr/local/include    -fPIC  -g -O2  -c MariaRow.cpp -o MariaRow.o
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/library/cpp11/include' -I'/usr/lib/R/library/plogr/include' -I/usr/local/include    -fPIC  -g -O2  -c MariaTypes.cpp -o MariaTypes.o
MariaTypes.cpp: In function 'MariaFieldType variable_type_from_field_type(enum_field_types, bool, bool)':
MariaTypes.cpp:43:8: error: 'MYSQL_TYPE_JSON' was not declared in this scope
   case MYSQL_TYPE_JSON:
        ^~~~~~~~~~~~~~~
MariaTypes.cpp:43:8: note: suggested alternative: 'MYSQL_TYPE_BLOB'
   case MYSQL_TYPE_JSON:
        ^~~~~~~~~~~~~~~
        MYSQL_TYPE_BLOB
make: *** [MariaTypes.o] Error 1
ERROR: compilation failed for package 'RMariaDB'
* removing '/home/fatkat/R/aarch64-unknown-linux-gnu-library/4.3/RMariaDB'

The downloaded source packages are in
	'/tmp/Rtmprph4qt/downloaded_packages'
Warning message:
In install.packages("RMariaDB") :
  installation of package 'RMariaDB' had non-zero exit status
>

Hello, I am facing the same error.
OS: CentOS7.9
R: 4.1.0
 rpm -qa | grep mariadb
mariadb-libs-5.5.68-1.el7.x86_64
mariadb-devel-5.5.68-1.el7.x86_64
mariadb-5.5.68-1.el7.x86_64
mariadb-server-5.5.68-1.el7.x86_64


MariaTypes.cpp: In function ‘MariaFieldType variable_type_from_field_type(enum_field_types, bool, bool)’:
MariaTypes.cpp:43:8: error: ‘MYSQL_TYPE_JSON’ was not declared in this scope; did you mean ‘MYSQL_TYPE_BLOB’?
   43 |   case MYSQL_TYPE_JSON:
      |        ^~~~~~~~~~~~~~~
      |        MYSQL_TYPE_BLOB
make: *** [MariaTypes.o] Error 1
ERROR: compilation failed for package ‘RMariaDB’
* removing ‘/root/R/x86_64-pc-linux-gnu-library/4.1/RMariaDB’

--- a/src/MariaTypes.cpp
+++ b/src/MariaTypes.cpp
@@ -40,7 +40,7 @@ MariaFieldType variable_type_from_field_type(enum_field_types type, bool binary,
   case MYSQL_TYPE_VAR_STRING:
   case MYSQL_TYPE_VARCHAR:
     return binary ? MY_RAW : MY_STR;
-  case MYSQL_TYPE_JSON:
+  case 245: // MYSQL_TYPE_JSON not defined in CentOS 7
     return MY_STR;
   case MYSQL_TYPE_BLOB:
   case MYSQL_TYPE_TINY_BLOB:

